src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesTreeRenderer.java

```diff
@@ -77,4 +81,20 @@ public void customizeCellRenderer(@NotNull JTree tree,
       setToolTipText(getText("ui.exercise.ExercisesTreeRenderer.doubleClickToOpenBrowser"));
     }
   }
+
+  /**
+   * Sometimes {@code customizeCellRenderer} is called with a value
+   * that is just some placeholder object for tree root.
+   * This method recognizes such cases.
+   *
+   * @param value A parameter passed to {@code customizeCellRenderer}
+   * @return True, if node is irrelevant and should be ignored, otherwise false.
+   */
+  private boolean isIrrelevantNode(Object value) {
+    // That irrelevant root has no user object
+    // We assume that all relevant nodes that implement DefaultMutableTreeNode
+    // do have user objects.
+    return value instanceof DefaultMutableTreeNode
+        && ((DefaultMutableTreeNode) value).getUserObject() == null;
+  }
```

Comment:
I have two questions related to this:

1. Is it always an instance of `DefaultMutableTreeNode`, i.e. can we remove the `instanceof`?
2. Our `TreeView` hides the root node, so I guess ideally it shouldn't be passed to the custom renderer at all, but I guess IntelliJ's `Tree` doesn't do that. Maybe a less weird way would be `isRoot`, which checks for `((DefaultMutableTreeNode) value).getParent() == null`

Reply:
1. Actual, relevant nodes could, in theory, be something else so it is good that the check is there
2. Hmm... so is it so, that for any relevant node it holds that their parent is never null? In that case, we should follow your suggestion. 👍

Reply:
1. Oops I missed that this is in `ExercisesTreeRenderer`, not `TreeView`. But in that case then this doesn't actually work if the tree model used is no longer `DefaultMutableTreeNode`, since the type of the root node would change as well, and it would no longer be classified as irrelevant. I guess we wouldn't rely on `DefaultMutableTreeNode` anywhere except in `TreeView` 
2. I'm pretty sure they should all have a parent, but I'll double-check to make sure.

Reply:
1. Yeah, the check is a bit fuzzy. It wasn't easy to find rules that could be assumed to always match to those "irrelevant" nodes, so I came up with that. If it's enough to check it ain't a root node then that's much better rule.

Reply:
(But I realize that this is just a fix for how `Tree` uses the custom renderer, so it doesn't have to nice or perfect)